components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -137,13 +139,16 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         String scimGroupsEndpoint;
         String scim2UsersEndpoint;
         String scim2GroupsEndpoint;
-        String samlAuthnRequestsSigningEnabled;
 
         openIdUrl = IdentityUtil.getProperty(IdentityConstants.ServerConfig.OPENID_SERVER_URL);
-        samlECPUrl = IdentityUtil.resolveURL(IdentityUtil.getProperty(IdentityConstants.ServerConfig.SAML_ECP_URL),
-                true, true);
-        samlArtifactUrl = IdentityUtil.resolveURL(IdentityUtil.getProperty(IdentityConstants.ServerConfig
-                .SSO_ARTIFACT_URL), true, true);
+        samlSSOUrl = IdentityUtil.getProperty(IdentityConstants.ServerConfig.SSO_IDP_URL) +
+                IdPManagementUtil.getTenantParameter();
```

Comment:
Should we add the tenant domain as a query param when the url is read from file?
@malithie @mefarazath @inthirakumaaran

Reply:
we should first read from the file. Then if it's empty we construct then only tenant query param should be appended